source/Sprite.h

```diff
@@ -60,14 +58,14 @@ class Sprite {
 	uint32_t Texture() const;
 	uint32_t Texture(bool isHighDPI) const;
 	// Get the collision mask for the given frame of the animation.
-	const Mask &GetMask(int frame = 0) const;
+	const Mask &GetMask(int frame = 0, double scale = 1.) const;
 	
 	
 private:
 	std::string name;
 	
 	uint32_t texture[2] = {0, 0};
-	std::vector<Mask> masks;
+	MaskManager *maskManager = nullptr;
```

Comment:
I can make this const by moving the mask clearing from Sprite::Unload to SpriteQueue::DoLoad, which is what calls Sprite::Unload, but as soon as I change this from storing a pointer to storing a reference it no longer seems to work.

If instead of having the Sprite store information about the MaskManager I change it so that the Body passes the MaskManager as a const reference when it asks for the mask (which already has access to the MaskManager via GameData for the use of adding scales) then it works, but at that point it seems like we could cut out the Sprite::GetMask method entirely and just have the Body in Body::GetMask ask the MaskManager for the relevant Mask.